### PR TITLE
feat: REST api's and Request Handler for Position Service

### PR DESCRIPTION
**Pull request by GregoryIvo**
This pr adds a new bundle `org.eclipse.kura.rest.position.provider`.  This class exposes `org.eclipse.kura.position` API's, to rest.

> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped

**Screenshots:** If applicable, add screenshots to help explain your solution

**Manual Tests**: Optional description of the tests performed to check correct functioning of changes, useful for an efficient review

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

**Comment by GregoryIvo** [comment]
## The Current REST API for review

#### Get Position
- Method: GET
- API PATH: `/services/position/v1/position/`

##### Responses
- 200 OK status

```JSON
{
	"longitude": 7.729571119959449,
	"latitude": 50.45345802194789,
	"altitude": 1000.0
}
```

- 400 Bad Request (Malformed Client JSON)
- 404 Resource Not Found
- 500 Internal Server Error
```JSON
{
	"message": "Service unavailable. The position is not locked."
}
```

---
#### Get Date Time
- Method: GET
- API PATH: `/services/position/v1/datetime/`

##### Responses
- 200 OK status

```JSON
{
	"localDateTime": "2023-07-19T18:26:38Z"
}
```

- 400 Bad Request (Malformed Client JSON)
- 404 Resource Not Found
- 500 Internal Server Error
```JSON
{
	"message": "Service unavailable. The position is not locked."
}
```

---
#### Get Is Position Locked
- Method: GET
- API PATH: `/services/position/v1/islocked/`

##### Responses
- 200 OK status

```JSON
{
	"localDateTime": "2023-07-19T18:26:38Z"
}
```

- 400 Bad Request (Malformed Client JSON)
- 404 Resource Not Found
- 500 Internal Server Error
```JSON
{
	"islocked": false
}
```

**Comment by GregoryIvo** [comment]
REST API
image
image
image

Request Handler API
image
image
image

File: kura/org.eclipse.kura.rest.position.provider/OSGI-INF/position_rest_service.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2021 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/OSGI-INF/position_rest_service.xml]:
```suggestion
    Copyright (c) 2023 Eurotech and/or its affiliates and others
```

File: kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/internal/rest/position/PositionRestService.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2022 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/internal/rest/position/PositionRestService.java]:
```suggestion
 * Copyright (c) 2023 Eurotech and/or its affiliates and others
```

File: kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/DateTimeDTO.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2022 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/DateTimeDTO.java]:
```suggestion
 * Copyright (c) 2023 Eurotech and/or its affiliates and others
```

File: kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/IsLockedDTO.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package org.eclipse.kura.rest.position.api;
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/IsLockedDTO.java:13]: Please add copyright header

File: kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/PositionDTO.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2022 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/PositionDTO.java]:
```suggestion
 * Copyright (c) 2023 Eurotech and/or its affiliates and others
```

File: kura/org.eclipse.kura.rest.position.provider/build.properties [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#
+#  Copyright (c) 2021 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/build.properties]:
```suggestion
#  Copyright (c) 2023 Eurotech and/or its affiliates and others
```

File: kura/org.eclipse.kura.rest.position.provider/pom.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2021, 2023 Eurotech and/or its affiliates and others
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/pom.xml]:
```suggestion
    Copyright (c) 2023 Eurotech and/or its affiliates and others
```

File: kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/PositionDTO.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+ *******************************************************************************/
+package org.eclipse.kura.rest.position.api;
+
+import org.osgi.util.position.Position;
+
+public class PositionDTO {
+
+    private final double longitude;
+    private final double latitude;
+    private final double altitude;
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/PositionDTO.java]: Should we also add the track and speed fields?

File: kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/PositionDTO.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.rest.position.api;
+
+import org.osgi.util.position.Position;
+
+public class PositionDTO {
+
+    private final double longitude;
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.rest.position.provider/src/main/java/org/eclipse/kura/rest/position/api/PositionDTO.java]: org.osgi.util.position.Position Javadoc reports that all of its fields can be null, maybe we should switch to Double to accomodate that and handle that case in the constructor. If we do that, we should also report it in the documentation

**Comment by GregoryIvo** [comment]
image

**Comment by GregoryIvo** [comment]
Docs PR can be found here: https://github.com/eclipse/kura/pull/4783

**Review by nicolatimeus — APPROVED** [review]

**Pull request merged**